Source/Assets/Scripts/PlayerMovementController.cs

```diff
@@ -9,12 +9,18 @@ public class PlayerMovementController : MonoBehaviour
 
     private void Update()
     {
+        
         GetMovementInput();
         MovePlayer();
     }
 
     private void GetMovementInput()
     {
+        if (PlayerInteractor.Instance.isInspecting)
```

Comment:
Rework this to use GetComponent<PlayerInteractor> in Awake() then use the instance. Let's not use Singletons unless absolutely necessary.